File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 6/12/2025, 8:28:11 PM
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit=0;
        int minvalue=INT_MAX;
        for(int i=0;i<prices.size();i++)
        {
           
                minvalue=min(prices[i],minvalue);
                profit=max(prices[i]-minvalue,profit);
            
        }
       return profit;
        
    }
};

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 23/07/2026, 15:08:27
class Solution(object):
    def maxProfit(self, prices):
        i=0
        r=len(prices)
        profit=0
        mini=999999
        while i<r:           
            profit=max(profit,prices[i]-mini)
            mini=min(mini,prices[i])
            i+=1
            
            
           
        return profit

            
        """
        :type prices: List[int]
        :rtype: int
        """
        